### DISCUSSION TITLE
Is there support for deep story sorting?

### DISCUSSION THREAD

**Discussion by noranda** [storybookjs/storybook#25765] [Help]
### Summary

I've been doing some digging and can't seem to find a solution. I'd like to be able to sort my deeply nested stories. For example:

```
- Home
- Usage
- Contribute
- Components
  - Button
    - Docs
    - Default
    - All Variants
```

When I try to use the guidance on sorting, it seems to have no affect on anything more than 2 levels deep. I've tried:

```ts
// preview.ts

const preview = {
  parameters: {
    // ...
    options: {
      storySort: {
        order: ['Home', 'Usage', 'Contribute', 'Components', ['*', ['Default', '*', 'All Variants']],
      },
    },
  },
};
```
However, while this works for my top level guides, it does not work for anything deeply nested. Is there a way to define a sort order for these stories?

**Comment by felix-tjernberg** [comment]
I also have this problem, @noranda did you figure it out?

My current solution is to sort alphabetically then add a _ to the stories that I want to come first which is quite hacky

For example

```
- components
  - some_component
    - _Default Story
    - _Usage Story
    - Some other Story
```

**Comment by smlimasu** [comment]
I just sort the stories in the stories file itself, like Default comes first and all variants last in the file, then it's the order in the sidebar. 

I am not sure if you could also sort them in the parameters of the stories meta instead of preview?

**Reply by felix-tjernberg** [reply]
Manual sorting, [aint nobody got time for that!](https://youtu.be/ydmPh4MXT3g?t=24)

It is [documented that you can only sort two levels deep _(described above the third example)_](https://storybook.js.org/docs/writing-stories/naming-components-and-hierarchy#sorting-stories), however it seems like a very arbitrary value set for number of levels

I'm just confused why they set this arbitrary limit without looking at their implementation, the thing that @noranda suggests seem perfectly reasonable and even expected behavior if you don't look at their docs

The other possibility would be to re-implement their functionality [making our own sorting functions _(shown in the first example)_](https://storybook.js.org/docs/writing-stories/naming-components-and-hierarchy#sorting-stories), however I don't have time to do that :(, so was hoping someone else had done this

If someone knows where in their source code they have implemented their sorting of stories by name function, someone could create a more general function by copying their implementation for us all to use as a custom sorting function (I'm to lazy to look for that xD)
